lib/plugins/cleaner.rb

```diff
@@ -0,0 +1,51 @@
+require 'rubygems/package'
+require_relative '../moonshot/creds_helper'
+
+module Moonshot
+  module Plugins
+    # Moonshot plugin class for removing artifacts when a stage gets deleted.
+    class Cleaner
+      include Moonshot::CredsHelper
+
+      def initialize(bucket_name)
+        @bucket_name = bucket_name
+      end
+
+      def post_delete(resources)
+        stack_name = resources.stack.name
+        resources.ilog.start 'Starting to delete artifacts for stack: ' \
+          "#{stack_name}" do |s|
+          arts = retrieve_artifact_names(stack_name)
+          if arts.empty?
+            s.success 'No artifacts found, nothing deleted.'
+          else
+            delete_artifacts arts
+            s.success "Deleted artifacts for stack: '#{stack_name}' " \
+                      "in bucket: '#{@bucket_name}'"
+          end
+        end
+      end
+
+      private
+
+      def retrieve_artifact_names(name)
+        objs = s3_client.list_objects(bucket: @bucket_name, prefix: name)
+        valid_keys = []
+        objs.contents.each do |o|
+          valid_keys << o.key if o.key =~ /^#{name}[\.-]/
+        end
+        valid_keys
+      end
+
+      def delete_artifacts(artifacts)
+        objects = artifacts.map { |a| { key: a } }
+        resp = s3_client.delete_objects(bucket: @bucket_name,
+                                        delete: { objects: objects } )
```

Comment:
Fair enough. :) But I wasn't only thinking internally. Like... I don't think it's healthy to save a 1000! artifacts for one stack. Like... I don't think that's a good idea in any case. Maybe if it's like, a long running stack and continuously deploy to it, which creates artifacts and you are running for a 100 days now. Okay then. :) 

Anyhow. Sure, I'll deal with it now rather than later. :)